### DISCUSSION TITLE
Prevent reserved variables to be passed on to the view

### DISCUSSION THREAD

**Discussion by LordSimal** [livewire/livewire#8003] [3. Feature ideas]
Don't know if this is livewire specific but lets see...

I was trying to output validation errors for a variable which worked without any problems in another component.

But was rather confused when I got the error:
```
Call to a member function getBag() on array
```
when doing a simple
```
@error('editingCategoryName')
    <x-input-error :messages="$message" class="mt-2"/>
@enderror
```

Reason behind that is the fact, that inside my `render()` method I did this:
```
$categories = Category::query()->with(['bankEntries'])
    ->orderBy('lft')
    ->get()
    ->toTree();
$hasErrors = Category::isBroken();
$treeErrors = $hasErrors ? Category::countErrors() : [];

return view('livewire.category.index', [
    'categories' => $categories,
    'hasErrors' => $hasErrors,
    'errors' => $treeErrors,
]);
```

So I unintentionally overwrote the `$errors` which is present by default.

My proposed solution would be to throw an exception if variables are passed on to the `view()` method if they contain keys which are known, reservered variables for default functionality.

**Comment by joshhanley** [comment]
@LordSimal yeah that's more likely a Laravel feature, as it's just a Laravel view object and Laravel handles the injection of the `$errors` bag.

**Reply by LordSimal** [reply]
Thanks for the quick reply 👍🏻

**Reply by joshhanley** [reply]
@LordSimal no worries! 🙂
